### PR TITLE
refactor: migrates to Chart.js

### PR DESCRIPTION
**Pull request by kleinfreund**
## Changes

Migrates from amcharts to Chart.js. As part of this change, the labels of the individual arcs have been removed (Chart.js doesn’t have built-in support for curving the labels along an arc’s curvature) and instead, a legend beneath the chart now labels the data.

The policies chart has been removed from the mesh overview page as it only ever represented one piece of data. The total count of policies is now shown beneath next to the Policies heading.

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

## Screenshots

<details><summary>Before:</summary>

Screenshot from 2023-01-25 10-57-54

Screenshot from 2023-01-25 10-59-22

</details>

<details><summary>After:</summary>

Screenshot from 2023-01-25 10-57-46

Screenshot from 2023-01-25 10-59-16

</details>

### CODE REVIEW

File: src/store/storeConfig.ts [hunk 1]
```diff
@@ -170,8 +171,18 @@ export const storeConfig: StoreOptions<State> = {
       meshInsightsFetching || serviceInsightsFetching || externalServicesFetching,
     getServiceResourcesFetching: ({ serviceInsightsFetching, externalServicesFetching }) =>
       serviceInsightsFetching || externalServicesFetching,
-    getChart: ({ overviewCharts }) => (chartName: string) => overviewCharts[chartName],
     getZonesInsightsFetching: ({ zonesInsightsFetching }) => zonesInsightsFetching,
+    getChart: (state) => {
+      return ({ chartName, title, subtitle = undefined, showTotal = false, isStatusChart = false }: { chartName: string, title: string, subtitle?: string, showTotal?: boolean, isStatusChart?: boolean }) => {
+        return {
```
Comment (johncowen) [inline src/store/storeConfig.ts:177]: bit: Could this be made a little easier to read, either multiline or I noticed you also have a new type for what looks like this in `index.d.ts`, could that be used here?

**Review by johncowen — COMMENTED** [review]
Looks good, I left a couple of tiny inline queries, then took it for a spin and its looks great!

I had a couple of non-important things, that could also be iterations later if they are a bit of effort, I think at least one is probably punt-able:

Screenshot 2023-01-26 at 15 05 48

^ Tooltips get cut off as the leave the canvas, guessing this is probably more that a bit of effort so maybe we just punt this one, unless its easier than I imagine.

Screenshot 2023-01-26 at 15 06 50

^ Sometimes it's hard to read the numbers in the tooltips, not sure if it has semi-transparency or something there, if this is more than a little effort I'd punt this also.

Lastly, I noticed the percentages aren't in the legends but are in the tooltips, should we add the percentages to the legends aswell?

File: package.json [hunk 2]
```diff
@@ -53,6 +54,7 @@
     "babel-jest": "^29.3.1",
     "babel-plugin-transform-import-meta": "^2.2.0",
     "babel-plugin-transform-vite-meta-env": "^1.0.3",
+    "canvas": "^2.11.0",
```
[the remark this answers is not in the document]
Comment (kleinfreund) [reply package.json:57]: Yep, that’s right. JSDom doesn’t implement canvas by default but will use it if the package is installed.

File: src/app/common/charts/DonutChart.vue [hunk 3]
```diff
@@ -1,442 +0,0 @@
...
-function createChart(): any {
-  if (!(chartNode.value instanceof HTMLElement)) {
-    return
-  }
-
-  chart = createFromConfig(createDefaultConfig(), chartNode.value, PieChart)
-
-  if (props.saveChart) {
-    // We declare these properties on globalThis in index.d.ts
-    // once these are gone we can also remove those declarations
```
[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/app/common/charts/DonutChart.vue:246]: Yes, good idea.

File: src/store/storeConfig.ts (hunk 1 above)
Comment (kleinfreund) [reply src/store/storeConfig.ts:177]: Right, I’ve changed this to use the type.

**Comment by kleinfreund** [comment]
> ^ Tooltips get cut off as the leave the canvas, guessing this is probably more that a bit of effort so maybe we just punt this one, unless its easier than I imagine.

I’ve made the font size smaller for tooltips, but it’ll generally be hard to fix this completely without more generously assigning fixed space to the charts.

> ^ Sometimes it's hard to read the numbers in the tooltips, not sure if it has semi-transparency or something there, if this is more than a little effort I'd punt this also.

This is fixed now.

> Lastly, I noticed the percentages aren't in the legends but are in the tooltips, should we add the percentages to the legends aswell?

I think this isn’t necessary.

**Review by johncowen — APPROVED** [review]
Cool ok, thanks for looking at those

**Review by lahabana — APPROVED** [review]
Looks great! Also very nice amount of code deletion :)

**Pull request merged**